casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -184,103 +184,102 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
+    "!C CX"     -> HadItsMatch(S0)(C_)(C_),
+    "!C C!"     -> HadItsMatch(S0)(C_)(C0 | S1),
+    "!C (C!)"   -> coveredBy("!C (!C)"),
+    "4X 4X"     -> SamePolarityMerge(F_)(F_)(Nil),
+    "4X 4!"     -> SamePolarityMerge(F0)(F_)(S1),
+    "4X CX"     -> SamePolarityMerge(F_)(C_)(Nil),
+    "4X C!"     -> SamePolarityMerge(F0)(C1)(S1),
+    "4X (!!4)"  -> PersistentCouldMatch(F_)(R0 | F_)(Nil),
+    "4! 4!"     -> CouldMatchSameConflicts(F_)(F_)(S0),
+    "4! 4!"     -> CouldMatchSameMerges(F0)(F1)(S0 | S1),
+    "4! CX"     -> SamePolarityMerge(F_)(C1)(S0),
+    "4! C!"     -> CouldMatchSameConflicts(F_)(C_)(S0),
+    "4! C!"     -> CouldMatchSameMerges(F0)(C1)(S0 | S1),
+    "CX CX"     -> SamePolarityMerge(C_)(C_)(Nil),
+    "CX C!"     -> CouldMatchSameConflicts(C0)(C0)(S0),
+    "CX C!"     -> CouldMatchSameMerges(C1)(C0)(S0),
+    "C! C!"     -> CouldMatchSameConflicts(C_)(C_)(S0),
+    "C! C!"     -> CouldMatchSameMerges(C0)(C1)(S0 | S1),
+    "CX !!X"    -> IncomingCouldMatch(R0)(C_)(Nil),
+    "(!4) !4"   -> coveredBy("!4 (!4)"),
+    "(!4) (!C)" -> coveredBy("(!4) !C"),
+    "(!4) (4!)" -> VolatileEvent(S0 | F_)(S0 | F_)(Nil),
+    "(!4) (C!)" -> PersistentNoMatch(S0 | F_)(C_ | S0)(Nil),
+    "(!4) 4X"   -> VolatileEvent(S0 | F_)(F_)(Nil),
+    "(!4) CX"   -> VolatileEvent(S0 | F_)(C_)(Nil),
+    "(!C) (!C)" -> PersistentNoMatch(S0 | C_)(S0 | C_)(Nil),
+    "(!C) (4!)" -> VolatileEvent(S0 | C_)(F_ | S0)(Nil),
+    "(!C) (C!)" -> PersistentNoMatch(S0 | C_)(C_ | S0)(Nil),
+    "(!C) 4!"   -> PersistentCouldMatch(S0 | C_)(F_)(S0),
+    "(!C) 4X"   -> PersistentNoMatch(S0 | C_)(F_)(Nil),
+    "(!C) C!"   -> PersistentCouldMatch(S0 | C_)(C_)(S0),
+    "(!C) CX"   -> PersistentNoMatch(S0 | C_)(C_)(Nil),
+    "(4!) (4!)" -> VolatileEvent(F_ | S0)(F_ | S0)(Nil),
+    "(4!) (C!)" -> VolatileEvent(F_ | S0)(C_ | S0)(Nil),
+    "(4!) 4!"   -> VolatileEvent(F1 | S1)(F_)(S0),
+    "(4!) C!"   -> VolatileEvent(F1 | S1)(C_)(S0),
+    "(C!) C!"   -> PersistentCouldMatch(C_ | S0)(C_)(S0),
+    "4! (4!)"   -> VolatileEvent(F_)(F1 | S1)(S0),
+    "4! (C!)"   -> PersistentCouldMatch(F_)(C_ | S0)(S0),
+    "4X (4!)"   -> VolatileEvent(F_)(F_ | S0)(Nil),
+    "4X (C!)"   -> PersistentNoMatch(F_)(C_ | S0)(Nil),
+    "C! (C!)"   -> PersistentNoMatch(C_)(C1 | S1)(S0),
+    "CX (C!)"   -> PersistentNoMatch(C_)(C_ | S0)(Nil),
+    "(!4) 4!"   -> VolatileEvent(S1 | F1)(F_)(S0),
+    "(!4) !C"   -> VolatileEvent(S1 | F1)(S0)(C0),
+    "(!4) C!"   -> VolatileEvent(S0 | F0)(C_)(S1),
+    "(4!) CX"   -> VolatileEvent(F_ | S0)(C_)(Nil),
+    "(C!) (C!)" -> PersistentNoMatch(C_ | S0)(C_ | S0)(Nil)
   )
 
   it should "respect mergeability rules when merging blocks" in effectTest {
     mergeabilityCases
       .map {
-        case ((name, desc), y) =>
-          val key = s"$name $desc"
-          y.attempt.map(_.fold(_ => s"$key FAILED", _ => key))
+        case (key, test) =>
+          test.attempt.map(_.fold(_ => s"$key FAILED", _ => key))
```

Comment:
I would rather add (very) short descriptions to all test cases

Reply:
How about threading the source position?

Reply:
How about using the source position and displaying it in the assertion?

Reply:
(are we doing that already? If so - case closed for me :))

Reply:
It's enough to revert back to:
```
it should "respect mergeability rules when merging blocks" in effectTest {
    mergeabilityCases.map(_._2).parSequence
  }
```

but then it will only display the first failure. I think this is good enough

Reply:
This is already the case with precondition failures